quiche/src/lib.rs

```diff
@@ -3857,8 +3857,10 @@ impl Connection {
         // Alternate trying to send DATAGRAMs next time.
         self.emit_dgram = !dgram_emitted;
 
-        // Create PING for PTO probe if no other ack-eliciting frame is sent.
-        if self.paths.get(send_pid)?.recovery.loss_probes[epoch] > 0 &&
+        // Create PING for PTO probe if no other ack-eliciting frame is sent or if
+        // we've sent too many non ACK eliciting packets without having
+        // sent an ACK eliciting one
+        if self.paths.get(send_pid)?.recovery.should_elicit_ack(epoch) &&
```

Comment:
https://github.com/cloudflare/quiche/blob/master/quiche/src/lib.rs#L3301 should be updated too I think?

Also, we can cache the value in a local `should_elicit_ack` variable or something, so we don't need to put this ultra-long method call in the condition itself (just for readability's sake).

Reply:
@ehaydenr I think the comment about https://github.com/cloudflare/quiche/blob/master/quiche/src/lib.rs#L3301 still needs addressing? Can just move the `should_elicit_ack` variable up and re-use it there too.